src/gsy_e/models/market/two_sided.py

```diff
@@ -344,15 +344,16 @@ def match_recommendations(
                     # re-raise exception to be handled by the external matcher
                     raise invalid_bop_exception
                 continue
-            original_bid_rate = market_bid.original_price / market_bid.energy
+            original_bid_rate = recommended_pair.bid_energy_rate - market_bid.accumulated_grid_fees
```

Comment:
AFAIU, you are subtracting a price from a rate, which is not possible due to different units. Can you please explain?

Reply:
Great catch, I've just fixed this.
Thanks!